### PR TITLE
RSDK-1755, RSDK-1872 Allow cloud-configured robot to work off-line using cached config

### PR DESCRIPTION
**Pull request by maximpertsov**
JIRA: 
* https://viam.atlassian.net/browse/RSDK-1755
* https://viam.atlassian.net/browse/RSDK-1872

This PR makes two changes to allow robots with cloud configs to run offline:
* Cache a robot's entire cloud config, including TLS information that was previously missing. If a cloud configured robot does not have TLS info, then it must make a request to viam.com to get new ones, otherwise the server crashes.
* Add timeout for the first time we create a gRPC client.

### CODE REVIEW

File: robot/impl/local_robot.go
Comment (penguinland) [inline robot/impl/local_robot.go:467]: I'm open to disagreement, but my usual preference is not to comment out code: just delete it entirely. Commented-out code tends to rot over time as things change around it. If we need this code again, it's in the git history, or someone can view this PR on github again.

[Example koan, from a website where every anecdote involves someone getting killed and that's not the interesting part of the story so don't get too distracted by that part](http://thecodelesscode.com/case/41)

File: components/arm/universalrobots/ur.go
Comment (penguinland) [inline components/arm/universalrobots/ur.go:245]: Why do these need to be added now, but not in the past?

File: config/reader.go
Comment (penguinland) [inline config/reader.go:502]: Again I'm open to disagreement, but my knee-jerk reaction is never to log the private key, even at the debug level. Maybe you intend to remove this line again before the PR is merged?

**Review by penguinland — COMMENTED** [review]
Thanks for figuring this out so quickly! I don't really know what I'm looking at here; please don't wait on my approval. but I was curious what the fix would be, so I looked anyway.

File: config/reader.go
Comment (maximpertsov) [reply config/reader.go:502]: yikes, yeah this one I meant to remove - good catch

```suggestion
```

File: web/server/net_logger.go
Comment (maximpertsov) [reply web/server/net_logger.go:250]: the main issue is that this log is pretty spammy if we are not connected to the internet, but maybe that means we need to be a little smarter about not making so many network logs if we suspect viam.com is unreachable

Comment (penguinland) [reply web/server/net_logger.go:250]: Yeah, that makes sense! One idea is to have a boolean containing whether the last log was successfully sent to the network, and only log this if it's true. Another idea is to leave it the way you've written it, but add in a comment warning folks that this can be super spammy. :)

File: robot/impl/local_robot.go
Comment (maximpertsov) [reply robot/impl/local_robot.go:466]: i generally agree - trying to see if there's a better way to do this. if not, we are probably going to make an additional comment that is linked to a ticket to revert these lines in a way that works.

File: web/server/net_logger.go
Comment (maximpertsov) [reply web/server/net_logger.go:250]: good idea. for the time being, i'm going to keep this at the info level just to reduce how much we are changing in this fix
```suggestion
			nl.loggerWithoutNet.Infof("error logging to network: %s", err)
```

File: config/reader.go
Comment (edaniels) [inline config/reader.go:558]:
```suggestion
```

**Review by edaniels — APPROVED** [review]

**Review by cheukt — APPROVED** [review]

**Pull request merged**
